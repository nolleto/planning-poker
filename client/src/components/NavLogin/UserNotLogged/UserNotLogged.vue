<template>
  <div>
    <a href="#" @click="$emit('sign-in')">Sign In</a>
    <span>or</span>
    <a href="#" @click="$emit('sign-up')">Sign Up</a>
  </div>
</template>

<script>
export default {
  name: 'UserNotLogged'
}
</script>
