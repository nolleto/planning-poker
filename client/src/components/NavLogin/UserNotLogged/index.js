export { default } from './UserNotLogged.vue'
