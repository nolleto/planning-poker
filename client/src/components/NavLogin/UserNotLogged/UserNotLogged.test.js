import UserNotLogged from './index'
import { shallowMount } from '@vue/test-utils'

const shallowComponent = () => {
  return shallowMount(UserNotLogged)
}

describe('UserNotLogged component', () => {
  let wrapper

  describe('Snapshots', () => {
    beforeAll(() => {
      wrapper = shallowComponent()
    })

    it('matches', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Event Listeners', () => {
    beforeAll(() => {
      wrapper = shallowComponent()
    })

    describe('When clicked in "Sign In" link', () => {
      beforeEach(() => {
        jest.spyOn(wrapper.vm, '$emit')
        wrapper.findAll('a').at(0).trigger('click')
      })

      it('emits "sign-in"', () => {
        expect(wrapper.vm.$emit).toHaveBeenCalledWith('sign-in')
      })
    })

    describe('When clicked in "Sign Up" link', () => {
      beforeEach(() => {
        jest.spyOn(wrapper.vm, '$emit')
        wrapper.findAll('a').at(1).trigger('click')
      })

      it('emits "sign-up"', () => {
        expect(wrapper.vm.$emit).toHaveBeenCalledWith('sign-up')
      })
    })
  })
})
