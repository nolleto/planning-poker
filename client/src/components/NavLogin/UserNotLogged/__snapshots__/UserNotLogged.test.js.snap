// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNotLogged component Snapshots matches 1`] = `
<div>
  <a
    href="#"
  >
    Sign In
  </a>
   
  <span>
    or
  </span>
   
  <a
    href="#"
  >
    Sign Up
  </a>
</div>
`;
